web/app/scripts/views/record/add-edit-controller.js

```diff
@@ -99,8 +152,32 @@
             ctl.editor.errors = validationErrors;
         }
 
+        function areConstantFieldsValid() {
+            /* jshint camelcase: false */
+
+            // basic validation for constant fields
+            if (!ctl.record.slug || !ctl.record.label || !ctl.record.geom ||
+                !ctl.record.geom.coordinates || ctl.record.geom.coordinates.length < 2 ||
+                !ctl.record.occurred_from || !ctl.record.occurred_to) {
+
+                $log.debug('Missing required constant field');
+                return false;
+            }
+
+            if (ctl.record.occurred_from > ctl.record.occurred_to) {
+                $log.debug('occurred from date cannot be later than occurred to date');
+                return false;
+            }
+            /* jshint camelcase: true */
+
+            return true;
+        }
+
         function onSaveClicked() {
-            if (ctl.editor.errors.length > 0) {
+            // set geom array back on record
+            ctl.record.geom.coordinates = [ctl.geom.lat, ctl.geom.lng];
```

Comment:
This won't work when adding a new record, because the geom object won't yet be defined.